inbox/models/thread.py

```diff
@@ -228,7 +228,10 @@ def mark_for_deletion(self):
         self.deleted_at = datetime.datetime.utcnow()
 
     discriminator = Column("type", String(16))
-    __mapper_args__ = {"polymorphic_on": discriminator}
+    __mapper_args__ = {
+        "polymorphic_on": discriminator,
+        "polymorphic_identity": "thread",
```

Comment:
newer SQLAlchemy versions required polymorphic_identity to be specified even on the base classes with SQLAlchemy inheritance. We never use the base polymorphic_identity value ourselves so the change from missing to `thread` does not affect us.